"Pick in Store" button always visible on checkout page...

nmalevanec opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. https://github.com/magento/inventory/issues/2647: "Pick in Store" button always visible on checkout page.
2. https://github.com/magento/inventory/issues/2650: Add to configuration of Store pickup method radius for search results of store selection

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. see https://github.com/magento/inventory/issues/2647

### For search radius
1. Create two sources with store pickup.(Culver City, Austin)
2. Create simple product and assign two sources to product.
3. Add product to cart and navigate to checkout.
4. Click 'Pick in Store' try to search Culver City. 
5. Verify, only Culver City source found.
6. Navigate to admin area Store>Configuration>Sales>Delivery Methods>In-Store Delivery
7. Change search radius to 2000 or greater
8. Navigate to checkout.
9. Click 'Pick in Store' try to search Culver City. 
10. Verify, both sources for Culver City and Austin are found.

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

nmalevanec committed:
MSI-2647: "Pick in Store" button always visible on checkout page.
MSI-2650: Add to configuration of Store pickup method radius for search results of store selection.
MSI-2653: If any of products has no inventory in particular source - it should not appear in Search result in Pickup store selection window

swnsma committed:
Merge branch 'store-pickup' into Store-PickUp-Checkout

nmalevanec committed:
MSI-2647: "Pick in Store" button always visible on checkout page.
MSI-2650: Add to configuration of Store pickup method radius for search results of store selection.
MSI-2653: If any of products has no inventory in particular source - it should not appear in Search result in Pickup store selection window.
Fix tests.

nmalevanec committed:
Merge branch 'store-pickup' into Store-PickUp-Checkout

sidolov commented:
@magento run all tests

sidolov committed:
Merge branch 'store-pickup' into Store-PickUp-Checkout

sidolov commented on app/code/Magento/InventoryInStorePickupFrontend/Block/Checkout/LayoutProcessor.php:
```diff
@@ -0,0 +1,204 @@
+<?php
```
Please, revert checkout layout xml and add only missing data in the LayoutProcessor

nmalevanec replied:
Done.

sidolov commented on app/code/Magento/InventoryInStorePickupFrontend/Model/Validator/StorePickUpValidator.php:
```diff
@@ -0,0 +1,80 @@
+ */
+class StorePickUpValidator
+{
+    const CONFIG_PATH = 'carriers/in_store/active';
```
Please, make const private

nmalevanec replied:
Done.

sidolov commented on app/code/Magento/InventoryInStorePickupFrontend/view/frontend/web/js/view/store-selector.js:
```diff
@@ -163,7 +163,13 @@ define([
+                skus = [],
+                items = quote.getItems();
+
+            _.each(items, function (item) {
```
Will it work with dynamic skus? For example configurable product with custom options?

nmalevanec replied:
Fixed

sidolov commented on app/code/Magento/InventoryInStorePickupFrontend/Model/Validator/StorePickUpValidator.php:
```diff
@@ -0,0 +1,80 @@
+/**
+ * Verify, if store pick-up available for given website.
+ */
+class StorePickUpValidator
```
What exactly this class validates? Could you please change to the name more explicit?

nmalevanec replied:
Renamed

nmalevanec committed:
MSI-2647: "Pick in Store" button always visible on checkout page.
MSI-2653: If any of products has no inventory in particular source - it should not appear in Search result in Pickup store selection window.
Code refactor. Add product with custom options support for distance based algorithm.

nmalevanec committed:
There is no need to filter store pickup available source by product skus.

nmalevanec committed:
Merge branch 'store-pickup' into Store-PickUp-Checkout

sidolov committed:
Merge branch 'store-pickup' into Store-PickUp-Checkout

sidolov committed:
Merge branch 'store-pickup' into Store-PickUp-Checkout

ishakhsuvarov committed:
magento/inventory#2661: Correctly remove store pickup components from layout

Commit:
magento/inventory#2661: Fix issue with undefined constant

Commit:
Merge remote-tracking branch 'msi/store-pickup' into nmalevanec-Store-PickUp-Checkout

sidolov commented:
@magento run Functional Tests CE

sidolov commented:
@magento run Unit Tests

sidolov commented:
@magento run Functional Tests CE,Unit Tests
